tasks/send-slack-notification/send-slack-notification.yaml

```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "release"
+  name: send-slack-notification
+spec:
+  description: >-
+    Sends message to slack using postMessage API
+  params:
+    - name: message
+      description: Message to be sent
+    - name: tasksStatus
+      description: status of tasks execution
+    - name: secretName
+      description: |
+        Name of secret which contains authentication token for app
+      default: slack-notification-secret
+  volumes:
+    - name: slack-token
+      secret:
+        secretName: $(params.secretName)
+  steps:
+    - name: send-message
+      image: >-
+        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+      volumeMounts:
+        - name: slack-token
+          mountPath: "/etc/secrets"
+          readOnly: true
+      env:
+        - name: HOME
+          value: /tekton/home
+        - name: MESSAGE
+          value: $(params.message)
+        - name: TASKSSTATUS
+          value: $(params.tasksStatus)
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        if [ -z "${MESSAGE/$'\n'}" ]; then
+          echo "Message is empty - skipping send to slack"
+          exit
+        fi
+
+        if [ -f "/etc/secrets/token" ]; then
+          TOKEN=$(cat etc/secrets/token)
+        else
+          echo "Secret not defined properly"
+          exit 1
+        fi
+
+        # replace @@CIRCLE_TYPE@@
+        # replace @@STATUS_TEXT@@
+        # set a default for circleType
```

Comment:
I have no idea what this stuff is. What is a circleType?

Reply:
circleType is the emoji that will be used in the message. We set the value based on the status of the pipeline. Red for Failure, Green for Succeeded. I will add a comment

Reply:
comment added

Reply:
I think that helps a lot, thank you